packages/lesswrong/server/resolvers/petrovDay2024Resolvers.ts

```diff
@@ -0,0 +1,51 @@
+import { PetrovDayLaunchs } from "@/lib/collections/petrovDayLaunchs";
+import { addGraphQLMutation, addGraphQLQuery, addGraphQLResolvers, addGraphQLSchema } from "../vulcan-lib";
+import { petrovDayLaunchCode } from "@/components/seasonal/PetrovDayButton";
+import PetrovDayActions from "@/lib/collections/petrovDayActions/collection";
+import { petrovBeforeTime } from "@/components/Layout";
+import { DatabaseServerSetting } from "../databaseSettings";
+import sample from "lodash/sample";
+
+const petrovFalseAlarmMissileCount = new DatabaseServerSetting<number[]>('petrovFalseAlarmMissileCount', [])
+const petrovRealAttackMissileCount = new DatabaseServerSetting<number[]>('petrovRealAttackMissileCount', [])
+
+const getIncomingCount = (incoming: boolean) => {
+  const currentHour = new Date().getHours();
+  const seed = currentHour + (incoming ? 24 : 0); // Different seed for each hour and incoming state
+
+  const missileCountArray = incoming ? petrovRealAttackMissileCount.get() : petrovFalseAlarmMissileCount.get();
+
+  return missileCountArray[seed % missileCountArray.length];
```

Comment:
I was going to say, this needs to be stable within each hour (else people could just keep refreshing and maybe figure out the different distributions that way).  Seems like it is stable now?